### PR TITLE
Fix npm for groups access tokens

### PR DESCRIPTION
**Pull request by RobiNino**
- [x] All [tests](https://github.com/jfrog/jfrog-cli-core#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] All [static analysis checks](https://github.com/jfrog/jfrog-cli-core/actions/workflows/analysis.yml) passed.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
-----
Since the `api/npm/auth` Artifactory endpoint doesn't handle groups Access Tokens well, since this PR we will avoid using this endpoint and construct the npm auth config manually for access tokens.
This will fix the issue for npm versions above `9.3.1` where scoped auth env vars are supported.
Tests passing in JFrog CLI: https://github.com/jfrog/jfrog-cli/pull/2604

### CODE REVIEW

File: artifactory/commands/yarn/yarn.go [hunk 1]
```diff
@@ -32,6 +32,8 @@ const (
 	//#nosec G101
 	yarnNpmRegistryServerEnv = "YARN_NPM_REGISTRY_SERVER"
 	yarnNpmAuthIndent        = "YARN_NPM_AUTH_IDENT"
+	// #nosec G101
+	yarnNpmAuthToken         = "YARN_NPM_AUTH_TOKEN"
 	yarnNpmAlwaysAuth        = "YARN_NPM_ALWAYS_AUTH"
```
Comment (yahavi) [inline artifactory/commands/yarn/yarn.go:37]: Please run go fmt
```suggestion
	yarnNpmAuthToken  = "YARN_NPM_AUTH_TOKEN"
	yarnNpmAlwaysAuth = "YARN_NPM_ALWAYS_AUTH"
```

File: artifactory/commands/npm/npmcommand_test.go [hunk 2]
```diff
@@ -61,40 +62,63 @@ func TestPrepareConfigData(t *testing.T) {
...
 		npmCm       *NpmCommand
+		authKey     string
 		value       string
 		expectedEnv string
 	}{
 		{
 			name: "set scoped registry auth env",
 			npmCm: &NpmCommand{
-				npmVersion: version.NewVersion("9.3.1"),
+				npmVersion: version.NewVersion("9.3.2"),
```
Comment (yahavi) [inline artifactory/commands/npm/npmcommand_test.go:80]: Why is that? It is better to test the edge case

File: artifactory/commands/npm/npmcommand_test.go [hunk 3]
```diff
@@ -61,40 +62,63 @@ func TestPrepareConfigData(t *testing.T) {
...
+			},
+			authKey:     utils.NpmConfigAuthKey,
+			value:       "some_auth_token",
+			expectedEnv: "npm_config__auth",
+		},
+		{
+			name: "set legacy auth env even though authToken is passed",
+			npmCm: &NpmCommand{
+				npmVersion: version.NewVersion("8.16.3"),
 				registry:   "https://registry.example.com",
```
Comment (yahavi) [inline artifactory/commands/npm/npmcommand_test.go:111]: The registry value repeats in all 4 tests. Let's set it in the loop.

**Review by yahavi — CHANGES_REQUESTED** [review]
